### PR TITLE
Fixed #35141 -- Clarified the expected type of CACHE_MIDDLEWARE_SECONDS setting.

### PR DESCRIPTION
**Pull request by e11bits**
This address ticket https://code.djangoproject.com/ticket/35141

The changes are as proposed in the ticket.

This is my fifth PR for Django so please advise as you see fit.

### CODE REVIEW

File: django/middleware/cache.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ class UpdateCacheMiddleware(MiddlewareMixin):
 
     def __init__(self, get_response):
         super().__init__(get_response)
-        self.cache_timeout = settings.CACHE_MIDDLEWARE_SECONDS
+        self.cache_timeout = int(settings.CACHE_MIDDLEWARE_SECONDS)
```
Comment (felixxm) [inline django/middleware/cache.py:70]: Can we clarify this is docs:
```diff
diff --git a/docs/topics/cache.txt b/docs/topics/cache.txt
index 880f606aae..ed0b9b3e24 100644
--- a/docs/topics/cache.txt
+++ b/docs/topics/cache.txt
@@ -608,8 +608,8 @@ entire site. You'll need to add
 Then, add the following required settings to your Django settings file:
 
 * :setting:`CACHE_MIDDLEWARE_ALIAS` -- The cache alias to use for storage.
-* :setting:`CACHE_MIDDLEWARE_SECONDS` -- The number of seconds each page should
-  be cached.
+* :setting:`CACHE_MIDDLEWARE_SECONDS` -- The integer number of seconds each
+  page should be cached.
 * :setting:`CACHE_MIDDLEWARE_KEY_PREFIX` -- If the cache is shared across
   multiple sites using the same Django installation, set this to the name of
   the site, or some other string that is unique to this Django instance, to

```

File: django/utils/cache.py [hunk 2]
```diff
@@ -411,7 +411,7 @@ def learn_cache_key(request, response, cache_timeout=None, key_prefix=None, cach
     if key_prefix is None:
         key_prefix = settings.CACHE_MIDDLEWARE_KEY_PREFIX
     if cache_timeout is None:
-        cache_timeout = settings.CACHE_MIDDLEWARE_SECONDS
+        cache_timeout = int(settings.CACHE_MIDDLEWARE_SECONDS)
```
Comment (felixxm) [inline django/utils/cache.py:414]: Can we add tests for these changes?

Comment (MiirzaBaig) [reply django/utils/cache.py:414]: Yes

Comment (felixxm) [reply django/utils/cache.py:414]: @MiirzaBaig Is there a reason why you leave useless comments?

**Comment by MiirzaBaig** [comment]
Sorry my bad.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: django/middleware/cache.py (hunk 1 above)
Comment (e11bits) [reply django/middleware/cache.py:70]: Of course

File: django/utils/cache.py (hunk 2 above)
Comment (e11bits) [reply django/utils/cache.py:414]: You are right

**Comment by e11bits** [comment]
I changed the PR as described in the ticket to just fix the docs for CACHE_MIDDLEWARE_SECONDS and the setting for that one test.

PS: I don't know how I managed to close the PR and had to reopen it. Sorry.

**Review by felixxm — COMMENTED** [review]
@e11bits Thanks :+1:

**Pull request merged**